### PR TITLE
Option in driving macro to do track matching w/ clusters

### PR DESCRIPTION
**Pull request by david-stewart**
May optionally generate a file with a TTree that recorded which track are matched (and which are not). May additionally put the clusteres of these tracks on the tree, along with which are, and are not, matched.

### CODE REVIEW

**Review by osbornjd — APPROVED** [review]
Here's a build which we can test just to be sure nothing changes, although it doesn't look like it will
https://web.sdcc.bnl.gov/jenkins-sphenix/job/sPHENIX/job/Build-Master-gcc12/1366/

File: detectors/sPHENIX/Fun4All_G4_sPHENIX.C [hunk 1]
```diff
@@ -321,6 +326,11 @@ int Fun4All_G4_sPHENIX(
   Enable::TRACKING_EVAL = Enable::TRACKING_TRACK && true;
   Enable::TRACKING_QA = Enable::TRACKING_TRACK && Enable::QA && true;
 
+  // only do track matching if TRACKINGTRACK is also used
```
Comment (pinkenburg) [inline detectors/sPHENIX/Fun4All_G4_sPHENIX.C:329]: This should be
Enable::TRACK_MATCHING = Enable::TRACKING_TRACK && false;
Enable::TRACK_MATCHING_TREE = Enable::TRACK_MATCHING && false;
Enable::TRACK_MATCHING_TREE_CLUSTERS = Enable::TRACK_MATCHING_TREE && false;
and remove the commented out lines 61-65. Then the user can activate this here (right after setting the tracking flags)

Comment (david-stewart) [reply detectors/sPHENIX/Fun4All_G4_sPHENIX.C:324]: That is very clever! The changes are made in this next push.

**Comment by osbornjd** [comment]
Thanks Chris, let's merge this

**Pull request merged**